src/Microsoft.Azure.WebJobs.Host/Scale/ScaleStatus.cs

```diff
@@ -12,5 +14,15 @@ public class ScaleStatus
         /// Gets or sets the current scale decision.
         /// </summary>
         public ScaleVote Vote { get; set; }
+
+        /// <summary>
+        /// Gets or sets the current target worker count.
+        /// </summary>
+        public int? TargetWorkerCount { get; set; }
+
+        /// <summary>
+        /// Gets the individual scale statuses for all monitored functions.
+        /// </summary>
+        public IDictionary<string, ScaleStatus> FunctionScaleStatuses { get; set; }
```

Comment:
The changes to this type don't look right. Note that in this member we're recursively referencing this type which isn't correct.

This type ScaleStatus as per its class doc is just the status type returned from IScaleMonitor. If you need a new type that represents aggregated status across monitors/scalers, we need a new type. In Functions [here](https://github.com/Azure/azure-functions-host/blob/379e98010c1d789369df6516b0eb5cc60556cdc7/src/WebJobs.Script/Scale/ScaleStatusResult.cs#L16) we had an aggregate type defined.

Reply:
You proposed the FunctionScaleStatuses in the commit:
https://github.com/Azure/azure-webjobs-sdk/pull/2954/commits/79a0486f4f7d18a902da55921771b9bb2b84f071#diff-bd390eccc6a95518edc67c13fef7a99bf7bfae7933b0a8ed4b6eb0180d5ae4b0R66

I do not think we want to introduce a new type, my previous proposal was to have 2 methods (see the commit diff):
```
public async Task<ScaleStatus> GetScaleStatusAsync(ScaleStatusContext context)
public async Task<IDictionary<string, ScaleStatus>> GetScaleStatusesAsync(ScaleStatusContext context)
```
What do you think?